tests/integration/kubernetes/k8s-guest-pull-image-signature.bats

```diff
@@ -0,0 +1,179 @@
+    setup_kbs_image_policy
+
+    create_pod_yaml_with_signed_image "${UNSIGNED_UNPROTECTED_REGISTRY_IMAGE}"
+
+    # For debug sake
+    echo "Pod ${kata_pod_with_signed_image}: $(cat ${kata_pod_with_signed_image})"
+
+    k8s_create_pod "${kata_pod_with_signed_image}"
+    echo "Kata pod test-e2e from image security policy is running"
+}
+
+@test "Create a pod from an unsigned image, on a 'restricted registry' is rejected" {
+    setup_kbs_image_policy
+
+    create_pod_yaml_with_signed_image "${UNSIGNED_PROTECTED_REGISTRY_IMAGE}"
+
+    # For debug sake
+    echo "Pod ${kata_pod_with_signed_image}: $(cat ${kata_pod_with_signed_image})"
+
+    assert_pod_fail "${kata_pod_with_signed_image}"
+    assert_logs_contain "${node}" kata "${node_start_time}" "Security validate failed: Validate image failed: Cannot pull manifest of ghcr.io/confidential-containers/test-container-image-rs:sha256-28e01ab32c9dbcbaae96cf0d5b472f22e231d9e603811857b295e61197e40a9b.sig"
+}
+
+@test "Create a pod from a signed image, on a 'restricted registry' is successful" {
+    setup_kbs_image_policy
+
+    create_pod_yaml_with_signed_image "${COSIGN_SIGNED_PROTECTED_REGISTRY_IMAGE}"
+
+    # For debug sake
+    echo "Pod ${kata_pod_with_signed_image}: $(cat ${kata_pod_with_signed_image})"
+
+    k8s_create_pod "${kata_pod_with_signed_image}"
+    echo "Kata pod test-e2e from image security policy is running"
+}
+
+@test "Create a pod from a signed image, on a 'restricted registry', but with the wrong key is rejected" {
+    setup_kbs_image_policy
+
+    create_pod_yaml_with_signed_image "${COSIGNED_SIGNED_PROTECTED_REGISTRY_WRONG_KEY_IMAGE}"
+
+    # For debug sake
+    echo "Pod ${kata_pod_with_signed_image}: $(cat ${kata_pod_with_signed_image})"
+
+    assert_pod_fail "${kata_pod_with_signed_image}"
+    assert_logs_contain "${node}" kata "${node_start_time}" "Security validate failed: Validate image failed: [PublicKeyVerifier"
+}
+
+@test "Create a pod from an unsigned image, on a 'restricted registry' works if enable_signature_verfication is false" {
+    setup_kbs_image_policy
+
+    create_pod_yaml_with_signed_image "${UNSIGNED_PROTECTED_REGISTRY_IMAGE}" "false"
+
+    # For debug sake
+    echo "Pod ${kata_pod_with_signed_image}: $(cat ${kata_pod_with_signed_image})"
+
+    k8s_create_pod "${kata_pod_with_signed_image}"
+    echo "Kata pod test-e2e from image security policy is running"
+}
+
+teardown() {
+    # if is_confidential_hardware; then
+    #     skip "Due to issues related to pull-image integration skip tests for ${KATA_HYPERVISOR}."
+    # fi
+
+    if ! is_confidential_runtime_class; then
+        skip "Test not supported for ${KATA_HYPERVISOR}."
+    fi
+
+    [ "${SNAPSHOTTER:-}" = "nydus" ] || skip "None snapshotter was found but this test requires one"
+
+    kubectl describe pods
+    k8s_delete_all_pods_if_any_exists || true
+
+    if [[ -n "${node_start_time}:-}" && -z "$BATS_TEST_COMPLETED" ]]; then
```

Comment:
s/"${node_start_time}:-}"/"${node_start_time:-}"/
Sorry - this was a mistake copied from my test